jmix-dynattr/dynattr/src/test/groovy/dynamic_attributes/DynAttrStringUtilsTest.groovy

```diff
@@ -0,0 +1,250 @@
+/*
+ * Copyright 2024 Haulmont.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package dynamic_attributes
+
+import io.jmix.dynattr.utils.DynAttrStringUtils
+import spock.lang.Specification
+
+class DynAttrStringUtilsTest extends Specification {
+    static char[] delimiters = [' ', '.', '_', '-', '\t'] as char[]
+
+    def "toCamelCase in PascalCase"() {
+        given:
+        String pascalCaseString = "PascalCaseString"
+
+        when:
+        String convertedCamelCase = DynAttrStringUtils.toCamelCase(pascalCaseString, delimiters)
+
+        then:
+        convertedCamelCase == "pascalCaseString"
+    }
+
+    def "toCamelCase in camelCase"() {
+        given:
+        String pascalCaseString = "camelCaseString"
+
+        when:
+        String convertedCamelCase = DynAttrStringUtils.toCamelCase(pascalCaseString, delimiters)
+
+        then:
+        convertedCamelCase == "camelCaseString"
+    }
+
+    def "toCamelCase in snake_case"() {
+        given:
+        String pascalCaseString = "snake_case_string"
+
+        when:
+        String convertedCamelCase = DynAttrStringUtils.toCamelCase(pascalCaseString, delimiters)
+
+        then:
+        convertedCamelCase == "snakeCaseString"
+    }
+
+    def "toCamelCase in kebab-case"() {
+        given:
+        String pascalCaseString = "kebab-case-string"
+
+        when:
+        String convertedCamelCase = DynAttrStringUtils.toCamelCase(pascalCaseString, delimiters)
+
+        then:
+        convertedCamelCase == "kebabCaseString"
+    }
+
+    def "toCamelCase in SCREAMING_SNAKE_CASE"() {
+        given:
+        String pascalCaseString = "SCREAMING_SNAKE_CASE_STRING"
+
+        when:
+        String convertedCamelCase = DynAttrStringUtils.toCamelCase(pascalCaseString, delimiters)
+
+        then:
+        convertedCamelCase == "screamingSnakeCaseString"
+    }
+
+    def "toCamelCase in space case"() {
+        given:
+        String pascalCaseString = "space case string"
+
+        when:
+        String convertedCamelCase = DynAttrStringUtils.toCamelCase(pascalCaseString, delimiters)
+
+        then:
+        convertedCamelCase == "spaceCaseString"
+    }
+
+    def "toCamelCase in half-capitalized Space case"() {
+        given:
+        String pascalCaseString = "Space case String"
```

Comment:
All such variables in all tests are called `pascalCaseString` that doesn't seem correct. Rename these variables.